src/super_gradients/common/code_analysis/breaking_change.py

```diff
@@ -0,0 +1,268 @@
+                            parameter_name=current_param,
+                            line_num=current_function_signature.line_num,
+                        )
+                    )
+
+        else:
+            # FunctionRemoved
+            breaking_changes.functions_removed.append(
+                FunctionRemoved(
+                    function_name=function_name,
+                    line_num=source_function_signature.line_num,
+                )
+            )
+
+    # Check import ONLY if __init__ file.
+    if module_path.endswith("__init__.py"):
+        source_imports = parse_imports(code=source_code)
+        current_imports = parse_imports(code=current_code)
+        breaking_changes.imports_removed = [
+            ImportRemoved(import_name=source_import, line_num=0) for source_import in source_imports if source_import not in current_imports
+        ]
+    return breaking_changes
+
+
+def analyze_breaking_changes(verbose: bool = 1) -> List[Dict[str, Union[str, List]]]:
+    """Analyze changes between the current branch (HEAD) and the master branch.
+    :param verbose: If True, print the summary of breaking changes in a nicely formatted way
+    :return:        List of changes, where each change is a dictionary listing each type of change for each module.
+    """
+    # GitHelper requires `git` library which should NOT be required for the other functions
+    from super_gradients.common.code_analysis.git_utils import GitHelper
+
+    root_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))
+    git_explorer = GitHelper(git_path=root_dir)
+
+    summary = ""
+    breaking_changes_list = []
+    for module_path in git_explorer.diff_files(source_branch="master", current_branch="HEAD"):
+
+        master_code = git_explorer.load_branch_file(branch="master", file_path=module_path)
+        head_code = git_explorer.load_branch_file(branch="HEAD", file_path=module_path)
+        breaking_changes = extract_code_breaking_changes(module_path=module_path, source_code=master_code, current_code=head_code)
+
+        if not breaking_changes.is_empty:
+            breaking_changes_list.append(breaking_changes.json())
+        summary += str(breaking_changes)
+
+    if verbose:
+        if summary:
+            print("{:<60} {:<8} {:<30} {}\n".format("MODULE", "LINE NO", "BREAKING TYPE", "DESCRIPTION (Master -> HEAD)"))
+            print("-" * 175 + "\n")
+            print(summary)
+        else:
+            print(colored("NO BREAKING CHANGE DETECTED!", "green"))
+
+    return breaking_changes_list
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Example script using flags")
+    parser.add_argument("--verbose", action="store_true", default=True, help="Enable verbose mode")
+    parser.add_argument("--output-file", default=None, type=str, help="Output file name")
+    parser.add_argument("--fail-on-error", action="store_true", default=True, help="Fail on error")
```

Comment:
i am not sure this works... 
action="store_true" is always default=False

Reply:
and if that is true, the default setting of this script is: do nothing. (no print, no file, no fail)

Reply:
You're right, I took it out 06f5db250bc210e19ed1e1bf0ea853cba27c16a3
Default value should not matter too much anyway because we set it explicitly in the CI